modules/markup/reStructuredText/reStructuredText_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package rst
+
+import (
+	"strings"
+	"testing"
+
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const AppURL = "http://localhost:3000/"
+const Repo = "go-gitea/gitea"
+const AppSubURL = AppURL + Repo + "/"
+
+func test(t *testing.T, input, expected string) {
+	buffer := RenderString(input, setting.AppSubURL, nil, false)
+	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(buffer))
+}
+
+func TestRender_StandardLinks(t *testing.T) {
+	var appURL = setting.AppURL
+	var appSubURL = setting.AppSubURL
+	setting.AppURL = AppURL
+	setting.AppSubURL = AppSubURL
+	defer func() {
+		setting.AppURL = appURL
+		setting.AppSubURL = appSubURL
+	}()
+
+	googleRendered := `<p><a href="https://google.com/">reStructuredText</a></p>`
+	test(t, "reStructuredText_\n\n.. _reStructuredText: https://google.com/\n", googleRendered)
+
+	// TODO: gorst didn't support relative link.
+	/*lnk := markup.URLJoin(AppSubURL, "WikiPage")
+	test("WikiPage_\n\n.. _WikiPage: WikiPage\n",
+		`<p><a href="`+lnk+`">WikiPage</a></p>`)*/
+}
+
+func TestRender_Images(t *testing.T) {
```

Comment:
If you don' need to change settings (look at `defer` comment above), you can merge both tests together, remove `setting` variable changes and `defer` and use `t.Run()` (just create array of test cases) to run tests in parallel.